lua/neogit/lib/functional.lua

```diff
@@ -0,0 +1,15 @@
+local util = require 'neogit.lib.util'
+local collect = require 'neogit.lib.collection'
```

Comment:
Initially I thought `collect` was a function, but this seems more like a constructor right? Could we PascalCase the variable if it is a constructor function?

Is the name supposed to be a short version of Collection or the verb?
```suggestion
local Collect = require 'neogit.lib.collection'
```

Reply:
The idea is that it's the verb. I'm kinda mixing paradigms here:  The structure is class-like, but the invocation is more like a builder pattern. We can totally change the initial invocation to `Collection:new()` if you feel that's more readable.

Reply:
> We can totally change the initial invocation to Collection:new() if you feel that's more readable.

I would really appreciate that! 👍 

> Collection:new()

Shouldn't it be `Collection.new()` ?

Reply:
Yeah it should. I just got stuck on that as well while pulling out that `output` thingy into a separate thing. I'm really not used to that operator :sweat_smile: